src/fastfeedparser/main.py

```diff
@@ -44,640 +47,573 @@ def parse(source):
-                enclosures.append(enc_item)
-
-        if enclosures:
-            entry["enclosures"] = enclosures
-
-        author = (
-            get_entry_value(
-                "author",
-                "{http://www.w3.org/2005/Atom}author/{http://www.w3.org/2005/Atom}name",
-                "{http://purl.org/dc/elements/1.1/}creator",
-            )
-            or get_entry_value(
-                "dc:creator",
-                "{http://purl.org/dc/elements/1.1/}creator",
-                "{http://purl.org/dc/elements/1.1/}creator",
-            )
-            or get_element_value(
-                item, "{http://purl.org/dc/elements/1.1/}creator", namespaces
-            )
-            or get_element_value(item, "author", namespaces)
+            thumb_item = {k: v for k, v in thumb_item.items() if v is not None}
+            if thumb_item:
+                media_contents.append(thumb_item)
+
+    if media_contents:
+        entry["media_content"] = media_contents
+
+    # Handle enclosures
+    enclosures: list[dict] = []
+    for enclosure in item.findall("enclosure"):
+        enc_item: dict[str, str | int | None] = {
+            "url": enclosure.get("url"),
+            "type": enclosure.get("type"),
+            "length": enclosure.get("length"),
+        }
+        # Convert length to integer if present and valid
+        length = enc_item["length"]
+        if length:
+            try:
+                enc_item["length"] = int(length)
+            except (ValueError, TypeError):
+                del enc_item["length"]
+
+        # Remove None values
+        enc_item = {k: v for k, v in enc_item.items() if v is not None}
+        if enc_item.get("url"):  # Only append if we have a URL
+            enclosures.append(enc_item)
+
+    if enclosures:
+        entry["enclosures"] = enclosures
+
+    author = (
+        get_field_value(
+            "author",
+            "{http://www.w3.org/2005/Atom}author/{http://www.w3.org/2005/Atom}name",
+            "{http://purl.org/dc/elements/1.1/}creator",
+            False,
         )
-        if author:
-            entry["author"] = author
-
-        if feed_type == "rss":
-            comments = get_element_value(item, "comments", namespaces)
-            if comments:
-                entry["comments"] = comments
-
-        return entry
+        or get_field_value(
+            "{http://purl.org/dc/elements/1.1/}creator",
+            "{http://purl.org/dc/elements/1.1/}creator",
+            "{http://purl.org/dc/elements/1.1/}creator",
+            False,
+        )
+        or _get_element_value(item, "{http://purl.org/dc/elements/1.1/}creator")
+        or _get_element_value(item, "author")
+    )
+    if author:
+        entry["author"] = author
+
+    if feed_type == "rss":
+        comments = _get_element_value(item, "comments")
+        if comments:
+            entry["comments"] = comments
+
+    return entry
+
+
+def _field_value_getter(
```

Comment:
Unified get field value across feed and item parsers, as well as moved the if checking logic outside the method to reduce overhead.